lock-api/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -78,4 +79,7 @@ default void tryUnlock(Set<LockToken> tokens) {
 
     long currentTimeMillis();
 
+    @Override
+    default void close() {}
```

Comment:
don't make this a default, otherwise we could end up skipping the close if we keep on wrapping.

Reply:
also, why do we have `TimeLockClient` and `RemoteTimelockServiceAdapter`, it's unclear what the distinction is between the two.

Reply:
TimelockClient takes a timelockService and adds few more higher level functionality, such as lock refreshing and autobatched unlocking. It is used for both embedded timelock and remote one.
`RemoteTimelockServiceAdapter` is only used for remote timelock, and converts `TimelockRpcClient` into `TimelockService`. Idea there is, timelock endpoints do not need to map one to one with timelock service methods because of perf optimisations. (Such as lock leases and batched start txn calls); and callers of `TimelockService` shouldn't care about these perf optimising endpoints, which are visible in `TimelockRpcClient` but not in `TimelockService`. Also these optimisations are only for remote timelock, not for embedded one.